client/asset/btc/btc.go

```diff
@@ -1755,18 +1790,24 @@ func fund(utxos []*compositeUTXO, enough func(uint64, uint64) bool) (
 			if len(okUTXOs) == 0 {
 				return false
 			}
-			// On each loop, find the smallest UTXO that is enough.
-			for _, txout := range okUTXOs {
-				if isEnoughWith(txout) {
-					addUTXO(txout)
-					return true
-				}
+
+			// Check if the largest output is too small.
+			lastUTXO := okUTXOs[len(okUTXOs)-1]
+			if !isEnoughWith(lastUTXO) {
+				addUTXO(lastUTXO)
+				okUTXOs = okUTXOs[0 : len(okUTXOs)-1]
+				continue
 			}
```

Comment:
Moving this check up up here was the critical improvement. Before, we iterated the utxos from smallest to largest to find the smallest utxo that satisfied our needs. If none worked, we'd add the last element and try again. But for `MaxOrder`, we're using all the utxos, so we'd run the enough function `N + (N - 1) + (N - 2) + ... + 1` = `N * (N + 1) / 2`, or roughly `N²/2`. With the new pattern, it's just `N`.